### ISSUE TITLE
please add cci

### ISSUE THREAD

**Issue by Jiraiyah** [bukosabino/ta#49]
@bukosabino 

Hi

Please can you add CCI (Commodity Channel Index) indicator too?

If you don't know, there are three different Momentum base formulas in TA : MACD, TSI, and CCI.

You already have two of them why not adding the last one please?

**Comment by Jiraiyah** [comment]
Oh I saw in documentation that you have it but i think there is something wrong at least on how i used to see the cci functions, for example, in trading view you can say cci(volume, 14) ! and yes it's a valid request to calculate cci on the volume (remember, it's a momentum based formula). but the way you implemented it, this would not work because volume has no high/low indices. not sure why you did this. even the cci for price in trading view is only asking the close value and period. maybe if you look into cci's documentation on trading view it can help?

https://www.tradingview.com/wiki/Commodity_Channel_Index_(CCI)

calculation section >>>>

There are several steps involved in calculating the Commodity Channel Index. 
The following example is for a typical 20 Period CCI:

CCI = (Typical Price  -  20 Period SMA of TP) / (.015 x Mean Deviation)

Typical Price (TP) = (High + Low + Close)/3

Constant = .015
The Constant is set at .015 for scaling purposes. 
By including the constant, the majority of CCI values will fall within the 100 to -100 range.

There are three steps to calculating the Mean Deviation. 
1. Subtract the most recent 20 Period Simple Moving from each typical price (TP) for the Period.
2. Sum these numbers strictly using absolute values. 
3. Divide the value generated in step 3 by the total number of Periods (20 in this case).

>>>>

looking at the above example, it's obvious why you are asking for high, and low, but.... it means that if i want to use the current implementation on your system, it would be enough to send volume three times as high and low !

then the question is..... why the function in trading view is not asking for high and low? you can easily say cci(close, 20) !!! it means that they are finding this high and low as highest and lowest? or.... when i experimented, i found something interesting !

they are IGNORING their own documentation !

when you say cci(close, 20) the thing they do themselves in the oscillator AND the function, is this :

cci = (close - sma(close, 20)) / (0.015 * mean deviation) !!! they are effectively asking YOU to choose the input yourself !

so... why not mimic the most reliable charting platform out there in net for this?

**Comment by Manser95** [comment]
+

**Comment by bukosabino** [comment]
I think this is the correct implementation. 

```python
    pp = (high + low + close) / 3.0
    mad = lambda x : np.mean(np.abs(x-np.mean(x)))
    cci = ((pp - pp.rolling(n, min_periods=0).mean())
           / (c * pp.rolling(n, min_periods=0).apply(mad, True)))
```

It is included in the new ta version `0.4.6`.

Thank you to all of you.
